clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -117,7 +117,7 @@ public void connecting(String id, long now, String host, ClientDnsLookup clientD
             connectionState.moveToNextAddress();
         } else {
             nodeState.put(id, new NodeConnectionState(ConnectionState.CONNECTING, now,
-                this.reconnectBackoffInitMs, ClientUtils.resolve(host, clientDnsLookup)));
+                this.reconnectBackoffInitMs, ClientUtils.resolve(host, clientDnsLookup), clientDnsLookup));
```

Comment:
Couldn't we pass the `host` to the constructor here as well and do the DNS resolution with the real host instead of trying to have the canonical hostname? (remove `getCanonicalHostName`)

Reply:
Yeah, It make sense!